packages/fuels-signers/src/provider.rs

```diff
@@ -399,6 +455,117 @@ impl Provider {
         self.client.produce_block(amount).await
     }
 
+    /// Get estimated cost of a transaction
+    pub async fn get_transaction_cost(
+        &self,
+        tx: &Transaction,
+        tolerance: Option<f64>,
+    ) -> Result<TransactionCost, Error> {
+        let NodeInfo {
+            min_gas_price,
+            min_byte_price,
+            ..
+        } = self.node_info().await?;
+
+        let tolerance = tolerance.unwrap_or(DEFAULT_GAS_ESTIMATION_TOLERANCE);
+        let dry_run_tx = Self::generate_dry_run_tx(tx);
+        let gas_used = self
+            .get_gas_used_with_tolerance(&dry_run_tx, tolerance)
+            .await?;
+
+        Self::calculate_tx_cost(
+            dry_run_tx,
+            tx.gas_price(),
+            tx.byte_price(),
+            min_gas_price.0,
+            min_byte_price.0,
+            gas_used,
+        )
+    }
+
+    fn generate_dry_run_tx(tx: &Transaction) -> Transaction {
+        let mut dry_run_tx = tx.clone();
+        // Simulate the contract call with MAX_GAS_PER_TX to get the complete gas_used
+        dry_run_tx.set_gas_limit(MAX_GAS_PER_TX);
+        dry_run_tx.set_gas_price(0);
+        dry_run_tx.set_byte_price(0);
+        dry_run_tx
+    }
```

Comment:
Functions are not just for code reuse but for code organization as well. Having this chunk named offers cognitive relief when reading the caller of this function who isn't interested in the details of the tx construction. As for the naming, I have no opinion on that, it seems ok. It being private limits the scope of the potential confusion.

Also, private functions are candidates for inlining since all of their calls are known at the time of compilation.